crates/ruff_linter/src/linter.rs

```diff
@@ -288,23 +288,30 @@ pub fn check_path(
         }
     }
 
-    // Remove fixes for any rules marked as unfixable.
-    for diagnostic in &mut diagnostics {
-        if !settings.rules.should_fix(diagnostic.kind.rule()) {
-            diagnostic.fix = None;
+    if parsed.is_valid() {
+        // Remove fixes for any rules marked as unfixable.
+        for diagnostic in &mut diagnostics {
+            if !settings.rules.should_fix(diagnostic.kind.rule()) {
+                diagnostic.fix = None;
+            }
         }
-    }
```

Comment:
I wasn't aware that we already had a similar loop. I guess this is okay. It does have the downside that we waste time on generating fixes that we then just omit. But I see why we do it this way, because we lack a better abstraction for testing if a fix should be generated.